src/main/java/org/opensearch/security/privileges/PrivilegesEvaluator.java

```diff
@@ -308,7 +309,14 @@ public PrivilegesEvaluatorResponse evaluate(
         }
 
         // Security index access
-        if (securityIndexAccessEvaluator.evaluate(request, task, action0, requestedResolved, presponse).isComplete()) {
+        if (securityIndexAccessEvaluator.evaluate(
+            request,
+            task,
+            action0,
+            requestedResolved,
+            presponse,
+            (ConfigModelV7.SecurityRoles) securityRoles
```

Comment:
I did a push with those changes. I've renamed the ConfigModelV6's and ConfigModelV7 inner classes to include V6/V7 in their name to make it clearer and avoid having to fully qualify them to differentiate from the interface.

The interface implementation is a bit rough still and could do with some cleanup, so i appreciate any feedback on that

Reply:
Just adding comment from today's call here, Nils said that  the 'V6' model refered to Elastic Search 6 and should'nt really be being used by anybody anymore, and suggested to deprecate it.

Reply:
We should not diverge the behavior if it can be avoided, switching from Generic -> Specific Impl is a big code change and its harder to review.

Lets not try to solve the V6 vs V7 issue on this pull request, what do you think about making a separate issue for follow up?